### DISCUSSION TITLE
Is NextJS good for dynamic content?

### DISCUSSION THREAD

**Discussion by SarahVanDenBerghe** [vercel/next.js#21220] [Help] [answered]
Hi, I'm a student at a university having previous experience with React, but not yet with NextJS! For a current uni assignment, I'm very interested in using NextJS because of the many benefits it has. However, I'd like to have some input from someone with more experience, since I'm not sure if NextJS in my current project is a good idea to use or not.

So in short, I'm making a React webapp with those features:
- Authentication system
- Users can create a project (will have its own detail page)
- Users can comment on each others project
- Users can like projects

Basically, the content of the website changes a lot, and I'm not sure if NextJS would be ideal since I assume that content would have to be server-side rendered every time, which I'm not sure is good in this case. It's a lot of dynamic content from users across the platform, so the database would change a lot. In theory, the project pages could be SSR, but since people can like and comment, it's not really a static webpage that I need.

Could someone share their experience or knowledge on this topic? Really curious to learn more!

**Comment by Ivano2000** [comment]
Follow

**Comment by vassbence** [comment, marked as answer]
Hell yeah it's good. Next.js can be a Create-React-App style SPA or a fully SSR website, or anything in between. For your use cases I think you should just write the pages and use client side fetching. It will be a good place to start. Next will pre-render these pages to plain html and on client side after React hydrates stuff will get fetched. Then later when you figured more stuff out about exactly what should be statically generated you can use some of the tools next gives you like `getStaticProps` etc

**Reply by SarahVanDenBerghe** [reply]
Oh good to hear! There are a few terms that I'm not familiar with but I'm sure I'll learn them along the way. Thank you for helping!
